modules/ROOT/pages/anypoint-custom-metrics-connector.adoc

```diff
@@ -19,6 +19,33 @@ include::partial$include-supported-mule-runtime-version.adoc[]
 
 You must also xref:am-installing.adoc[install the Anypoint Monitoring Agent].
 
+== Custom Metrics Best Practices
+
+=== Use Time Period and Frequency Effectively
+
+Each chart in the Custom Metrics dashboard represents the output of a query. Two parameters in the dashboard, _time range_ and _refresh frequency_, affect the response time for the query that generates a chart.
+
+Time range:: determines how much data is fetched or processed to generate the chart. For example, more data is fetched and processed when displaying a monthly chart compared to displaying a daily chart.
+
+Refresh frequency:: determines how often a chart refreshes. High refresh frequencies put more stress on a system and can affect performance. For example, a refresh rate of one hour sends a query every hour, and a refresh rate of one minute sends the same query every minute. 
+
+Increasing time range (such as daily to monthly) or increasing refresh frequency (from one hour to one minute) may start affecting the system response time. Reduce time range or reduce the refresh frequency to improve the chart response time.
+
+=== Use Low Cardinality Dimensions
+
+Most custom metric dimensions should have less than 100 possible unique values, but some may have a thousand unique values, such as `country_code`, `state_code`, `city_code`, `zip`, and so on. You should not add _high cardinality_ data, or data with many unique values, to custom metrics. Some examples of high cardinality dimensions include `customer_id`, `user_id`, `session_id`, `order_id`, `transaction_id`, `email_address`, `token_id`, `product_id`, `location_id` (latitude/longitude), `ip_address`, `phone_number`, and so on. Charts configured for custom metrics with high cardinality dimensions can adversely affect the response time of dashboards. For best results, restrict dimensions to have a only a few unique values.
+
+If a particular custom metrics chart starts slowing down and affecting other charts in the system, you can:
+
+* Create a new custom metric without the offending high cardinality dimension
+* Replace a high cardinality dimension with a low cardinality dimension, such as:
+** `category_id` instead of `product_id` 
+** `country_code` instead of `location_id`
+** `order_count` instead of `order_id`
+* Avoid using UUIDs and timestamps when creating new custom metrics
+* Reduce the time range of the custom metric chart, for example, creating a daily or weekly chart instead of a monthly or yearly chart 
```

Comment:
```suggestion
* Reduce the time range of the custom metric chart, for example, create a daily or weekly chart instead of a monthly or yearly chart 
```